### PR TITLE
Add submodule support

### PR DESCRIPTION
**Pull request by mpanarin**
This PR adds a support for submodule add command as they also support the `--reference` parameter

### CODE REVIEW

File: git_autoshare/clone.py [hunk 1]
```diff
@@ -8,31 +8,36 @@
...
 
 def main():
     cmd = [git_bin(), 'clone'] + sys.argv[1:]
-    skip = \
-        '--reference' in cmd or \
-        '--reference-if-able' in cmd or \
-        '-s' in cmd or \
-        '--share' in cmd
+    skip = any(
+        [c in cmd for c in [
```
Comment (guewen) [inline git_autoshare/clone.py]: you don't need the list comprehension as `any` can take a generator
```suggestion
        c in cmd for c in [
```

File: git_autoshare/clone.py [hunk 2]
```diff
@@ -8,31 +8,36 @@
...
+            '--reference',
+            '--reference-if-able',
+            '-s',
+            '--share',
+        ]]
```
Comment (guewen) [inline git_autoshare/clone.py]:
```suggestion
        ]
```

**Comment by sbidoul** [comment]
Oh, contributors. Cool.

**Comment by sbidoul** [comment]
Thanks for this contrib.

Interestingly, git-autoshare was originally made so `pip install git+https://some.very.big.repo`  would be fast and therefore avoid the need for submodules.

I'm happy to merge this nevertheless.

I had a couple of commits that where pending that I just pushed. Unfortunately https://github.com/acsone/git-autoshare/commit/c3e624492fa32fcdc899e9daa34ee401c052d3ac creates a small conflict (sorry about that). @mpanarin Could you rebase and resolve it?

I'll then use it a couple of days to check there is no negative side effects then merge.

**Comment by mpanarin** [comment]
@sbidoul rebased and ready to be tested )

I'll also look into the `git submodule update --init` as suggested by @yvaucher

File: git_autoshare/core.py [hunk 3]
```diff
@@ -124,3 +218,8 @@ def prefetch_one(host, orgs, repo, repo_dir, private, quiet):
 def prefetch_all(quiet):
     for host, orgs, repo, repo_dir, private in repos():
         prefetch_one(host, orgs, repo, repo_dir, private, quiet)
+
+
+if __name__ == '__main__':
+    import pdb; pdb.set_trace()
```
[the remark this answers is not in the document]
Comment (mpanarin) [reply git_autoshare/core.py]: fixed :D

**Comment by mpanarin** [comment]
@sbidoul @yvaucher the `submodule update` will be added in a separate PR as it is unstable so not to mess with this stable change

**Comment by sbidoul** [comment]
@Tonow-c2c I don't understand your last comment, this PR has nothing GitHub specific, it's pure git stuff.

**Comment by Tonow-c2c** [comment]
@sbidoul sorry I confused between git and github

**Comment by sebalix** [comment]
I configured `git-autoshare` and run a `git autoshare-prefetch`.

Now I want to clone a repository with submodules, but the following command doesn't clone from the cache:

    $ git-autoshare-clone --recurse-submodules git@github.com:owner/repo.git

Is this command supported?

**Pull request merged**